### PR TITLE
Fixes for R

### PR DESCRIPTION
**Pull request by haampie**
Edit: initially this PR modified libmkl things incorrectly, that's now removed.

- r: add libtirpc
- r: always use external blas/lapack

### CODE REVIEW

**Comment by rscohn2** [comment]
`-lmkl_gf` is for gfortran and `-lmkl_intel` is for gcc:

I am checking with the MKL team on what to do for applications where C & fortran need to call MKL.

```
(cvenv) rscohn1@anpfclxlin02:dr$ mkl_link_tool -libs gfortran

    Intel(R) oneAPI Math Kernel Library (oneMKL) Link Tool v6.3
    ===========================================================

Output
======

Linking line:
 -L${MKLROOT}/lib/intel64 -Wl,--no-as-needed -lmkl_gf_lp64 -lmkl_intel_thread -lmkl_core -liomp5 -lpthread -lm -ldl
(cvenv) rscohn1@anpfclxlin02:dr$ mkl_link_tool -libs gcc

    Intel(R) oneAPI Math Kernel Library (oneMKL) Link Tool v6.3
    ===========================================================

Output
======

Linking line:
 -L${MKLROOT}/lib/intel64 -Wl,--no-as-needed -lmkl_intel_lp64 -lmkl_intel_thread -lmkl_core -liomp5 -lpthread -lm -ldl
(cvenv) rscohn1@anpfclxlin02:dr$
```

**Comment by haampie** [comment]
Confusing :) is this about `daxpy_` vs `daxpy`? Don't most C projects add a `_` suffix through configure tests? Or passing values vs pointers?

**Comment by rscohn2** [comment]
Here is the answer from MKL team:

> The main difference between mkl_intel_lp64 and mkl_gf_lp64 is a few BLAS DOT functions with GNU Fortran specific ABI. All other functions are the same.
> So there is two possible options:
> 1.	If BLAS DOT functions are not used or used in one type of libraires you can use one oneMKL interface based on where this functionality is used.
> 2.	If BLAS DOT functions are used in both C and Fortran libraires that the application includes, unfortunately we can’t simply link both oneMKL interface libraries, instead oneMKL libraires should be loaded into local namespace for each C or Fortran library, so that C or Fortran library could resolve its own version of oneMKL interface library.

MKL defines `FOO` and `FOO_` so underscores are not usually a problem. This issue does not apply to intel fortran and is specific to a small subset of the MKL API. It doesn't seem like spack has enough information to choose between the options. Is there a way for spack to make the choice explicit like a variant?

**Comment by haampie** [comment]
Right, R is checking for a symbol `zdotu` used in fortran code, but ultimately links with gcc, hence my thinking that the `g` in `gf` was for gnu / GCC in general.

File: var/spack/repos/builtin/packages/r/package.py [hunk 1]
```diff
@@ -60,21 +59,20 @@ class R(AutotoolsPackage):
     version("3.1.3", sha256="07e98323935baa38079204bfb9414a029704bb9c0ca5ab317020ae521a377312")
     version("3.1.2", sha256="bcd150afcae0e02f6efb5f35a6ab72432be82e849ec52ce0bb89d8c342a8fa7a")
 
-    variant(
-        "external-lapack", default=False, description="Links to externally installed BLAS/LAPACK"
-    )
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/r/package.py:65]: :+1:  If it is external, it should be in `packages.yaml`

**Review by alalazo — APPROVED** [review]

**Pull request merged**